panic: semaphore: released more than held

This panic happened several times during the few minutes we had pipelining enabled

```
panic: semaphore: released more than held

goroutine 16137 [running]:
golang.org/x/sync/semaphore.(*Weighted).Release(0xc0124cfd60, 0x1?)
	/Users/derek/go/pkg/mod/golang.org/x/sync@v0.1.0/semaphore/semaphore.go:103 +0xbc
github.com/buraksezer/olric.(*DMapPipeline).Exec.func1()
	/Users/derek/go/pkg/mod/github.com/buraksezer/olric@v0.5.0/pipeline.go:424 +0x87
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/Users/derek/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
```

--- a/pipeline.go
+++ b/pipeline.go
@@ -414,7 +414,10 @@ func (dp *DMapPipeline) Exec(ctx context.Context) error {
 	numCpu := 1
 	sem := semaphore.NewWeighted(int64(numCpu))
 	for i := uint64(0); i < dp.dm.clusterClient.partitionCount; i++ {
-		_ = sem.Acquire(ctx, 1)
+		err := sem.Acquire(ctx, 1)
+		if err != nil {
+			return err
+		}
 
 		partID := i
 		errGr.Go(func() error {

